airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/test/util/destination_process/DockerizedDestination.kt

```diff
@@ -4,85 +4,214 @@
+                "--rm",
+                "--init",
+                "-i",
+                "-w",
+                "/data/job",
+                "--log-driver",
+                "none",
+                "--name",
+                containerName,
+                "--network",
+                "host",
+                "-v",
+                String.format("%s:%s", workspaceRoot, "/data"),
+                "-v",
+                String.format("%s:%s", localRoot, "/local"),
+                "-e",
+                "DEPLOYMENT_MODE=$testDeploymentMode",
+                // Yes, we hardcode the job ID to 0.
+                // Also yes, this is available in the configured catalog
+                // via the syncId property.
+                // Also also yes, we're relying on this env var >.>
+                "-e",
+                "WORKER_JOB_ID=0",
+                imageName,
+                command,
+            )
 
-    override fun shutdown() {
-        // close stdin, wait until process exits
-        TODO("Not yet implemented")
-    }
-}
+        fun addInput(paramName: String, fileContents: Any) {
+            Files.write(
+                jobRoot.resolve("destination_$paramName.json"),
+                Jsons.writeValueAsBytes(fileContents)
+            )
+            cmd.add("--$paramName")
+            cmd.add("destination_$paramName.json")
+        }
+        config?.let { addInput("config", it) }
+        catalog?.let { addInput("catalog", it) }
 
-// This is currently unused, but we'll need it for the Docker version.
-// it exists right now b/c I wrote it prior to the CliRunner retooling.
-/**
- * There doesn't seem to be a built-in equivalent to this? Scanner and BufferedReader both have
- * `hasNextLine` methods which block until the stream has data to read, which we don't want to do.
- *
- * This class simply buffers the next line in-memory until it reaches a newline or EOF.
- */
-private class LazyInputStreamReader(private val input: InputStream) {
-    private val buffer: ByteArrayOutputStream = ByteArrayOutputStream()
-    private var eof = false
+        logger.info { "Executing command: ${cmd.joinToString(" ")}" }
+        process = ProcessBuilder(cmd).start()
 
-    /**
-     * Returns the next line of data, or null if no line is available. Doesn't block if the
-     * inputstream has no data.
-     */
-    fun nextLine(): MaybeLine {
-        if (eof) {
-            return NoLine.EOF
-        }
-        while (input.available() != 0) {
-            when (val read = input.read()) {
-                -1 -> {
-                    eof = true
-                    val line = Line(buffer.toByteArray().toString(Charsets.UTF_8))
-                    buffer.reset()
-                    return line
-                }
-                '\n'.code -> {
-                    val bytes = buffer.toByteArray()
-                    buffer.reset()
-                    return Line(bytes.toString(Charsets.UTF_8))
+        runBlocking {
```

Comment:
not sure what best practice is here - is it better to just make this function a suspend fun?